WalletWasabi.Fluent/ViewModels/Wallets/Buy/Workflows/Workflow.cs

```diff
@@ -144,11 +147,16 @@ private void BindCurrentStepConversation()
 		// Dereference of a possibly null reference.
 		// Reason: this warning is dubious here.
 		// The parameter of WhenAnyValue() is an Expression (from System.Linq.Expressions).
-		// It's not directly executable code and therefore it cannot raise a NullReferenceException
+		// It's not directly executable code, therefore it cannot raise a NullReferenceException
 		// Also, null propagation isn't allowed by the compiler inside such an Expression,
 		// so the only way to remove this warning is to make CurrentStep non-nullable, which doesn't make sense by design.
 		this.WhenAnyValue(x => x.CurrentStep.Conversation)
-			.BindTo(this, x => x.Conversation);
+			.BindTo(this, x => x.Conversation)
+			.DisposeWith(_disposables);
 #pragma warning restore CS8602 // Dereference of a possibly null reference.
 	}
+
+	// TODO: Workflow holds subscriptions to other objects it doesn't own, so it must be disposable.
+	// Please, review when to dispose it to avoid memory leaks.
```

Comment:
I added to OrderViewModel.Dispose

`Workflow.Dispose();`

Should be fine, although I'm hesitant about having a client instance call the `Dispose` method on dependencies that have been provided through constructor injection. It strikes me as a questionable practice for an instance to directly manage the lifecycle of a dependency that could be shared with other clients, potentially leading to conflicts. If there are no more elegant alternatives, I suppose we'll have to make do. However, it would be ideal if @ichthus1604, who devised the workflow design in the end, could review it and suggest a more fitting solution.

Reply:
@SuperJMN 

The considerations you've raised are very reasonable to me.

`Workflow` is currently instantiated by `BuyViewModel` [here](https://github.com/SuperJMN/WalletWasabi/blob/improvements/explicit-disposal/WalletWasabi.Fluent/ViewModels/Wallets/Buy/BuyViewModel.cs#L163) and [here](https://github.com/SuperJMN/WalletWasabi/blob/improvements/explicit-disposal/WalletWasabi.Fluent/ViewModels/Wallets/Buy/BuyViewModel.cs#L193).

I think we could move that code into the `OrderViewModel` constructor, making it responsible for the entire lifetime of the `Workflow` instance, as a "child" of OrderViewModel, instead of using ctor injection.

Also, since we already have a factory method `Workflow.Create()`, we could move the empty order logic inside that.

I'm writing a PR which does this.

Reply:
@SuperJMN update:

I've just realized this isn't as trivial as it seemed initially.

I'll write a separate PR with a small refactoring, which can hopefully get merged into master so you can later grab it on here and continue from that.